pkg/events/event_system.go

```diff
@@ -38,20 +42,57 @@ type EventSystem interface {
 }
 
 type EventSystemImpl struct {
-	Store     *EventStore // storing eventChannel
-	publisher *EventPublisher
+	eventSystemId string
+	Store         *EventStore // storing eventChannel
+	publisher     *EventPublisher
 
 	channel chan *si.EventRecord // channelling input eventChannel
 	stop    chan bool            // whether the service is stop
 	stopped bool
 
-	sync.Mutex
+	trackingEnabled bool
+	requestsEnabled bool
+	requestCapacity int
+	bufferCapacity  int
+	responseMaxSize int
```

Comment:
Consider adding comments to these entries. If "bufferCapacity" refers to the ring buffer, then rename it to "ringBufferCapacity" to make it clear.

Reply:
PR updated with the changes.